### ISSUE TITLE
GUI TrackWidget::TrackWidget constructor use of virtual function

### ISSUE THREAD

**Issue by tsteven4** [GPSBabel/gpsbabel#573]
clang-tidy-9 finds
```
filterwidgets.cc:96:3: warning: Call to virtual function during construction [clang-analyzer-optin.cplusplus.VirtualCall]
  checkChecks();
  ^
filterwidgets.cc:96:3: note: This constructor of an object of type 'TrackWidget' has not returned when the virtual method was called
filterwidgets.cc:96:3: note: Call to virtual function during construction
1 warning generated.
```

clang-tidy-10 puts it a little differently:

```
filterwidgets.cc:96:3: warning: Call to virtual method 'TrackWidget::checkChecks' during construction bypasses virtual dispatch [clang-analyzer-optin.cplusplus.VirtualCall]
  checkChecks();
  ^
filterwidgets.cc:96:3: note: Call to virtual method 'TrackWidget::checkChecks' during construction bypasses virtual dispatch
```

```

**Comment by tsteven4** [comment]
It looks like TrackWidget::checkChecks() should be marked final instead of virtual.

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final

**Comment by tsteven4** [comment]
This was fixed by #1387
